core/client/fs/src/main/java/alluxio/client/file/cache/PageStore.java

```diff
@@ -114,54 +115,61 @@ static void initialize(PageStoreOptions options) throws IOException {
   /**
    * Writes a new page from a source channel to the store.
    *
-   * @param pageId page identifier
-   * @param page page data
+   * @param pageId   page identifier
+   * @param page     page data
+   * @param pageInfo page info
    * @throws ResourceExhaustedException when there is not enough space found on disk
-   * @throws IOException when the store fails to write this page
+   * @throws IOException                when the store fails to write this page
    */
-  void put(PageId pageId, byte[] page) throws ResourceExhaustedException, IOException;
+  void put(PageId pageId, byte[] page, PageInfo pageInfo)
```

Comment:
since PageId is contained in PageInfo, does it make sense to have the following signature?
In general we prefer simpler and more concise code.

```suggestion
  void put(PageInfo pageInfo, byte[] page)
```